### ISSUE TITLE
repo naming?

### ISSUE THREAD

**Issue by aguestuser** [crisisbox/docker#4]
i am confused about the naming of this repo. (not necessarily because it is ill-named: potentially because i am thinking about it wrong!)

to wit: by conventions discussed today, the docker config for a given service should live in the repo corresponding to that service. if i inspect the `docker-compose.yml` file, i see that the docker image(s) it specifies just spin up everything needed to run owncloud behind an nginx reverse proxy.

so... wouldn't this repo be more appropriately named `owncloud`? and if so... should we observe the convention that the docker files that live with it are only those necessary to run the owncloud service?

cc: @gmarceau @avmcrae @aepyornis

**Comment by avmcrae** [comment]
Yes I would agree. Docker here is the tool/implementation detail, and I feel like the repo name should be based on what it is trying to do. 

And yes, repo-specific Dockerfiles should live with the repo itself! Here there is one dockerfile for both intake-form and intake-receiver, but if we are planning to combine them into one repo anyway that makes sense.

**Comment by avmcrae** [comment]
Also, just noticing in the intake dockerfile... we should probably add babel-cli to the package.json rather than having it as a global dependency

**Comment by aepyornis** [comment]
I agree that repo specific docker files should with the code corresponding to that service.

Regarding if this should be called "owncloud", I think calling it owncloud still doesn't explain everything that it does. Right now, the docker-compose file launches three services:

- Owncloud
- Our intake form
- Nginx to proxy to those services

I think this repo should also contain a 'production' docker build script that will generate an nginx configuration with correct host + tls certs for deployment.  As the project grows, won't it soon include more components than just owncloud? I'm not necessarily saying it should be called simply _docker_ either.

**Comment by aguestuser** [comment]
So it sounds like the purpose of the code you have written is to orchestrate the build and deployment of *all* of our services. I propose moving whatever code has that purpose to the repo newly renamed from `composer` -> `build`.

Only thing I'm not really sure about (because I'm not sure how you were thinking about composing things) is how you intend for the docker compose file to compose dockerfiles that might live in separate repos that the `build` repo might know about.

To give a hint at how I'm envisioning this working in a pipeline, check out the logic in this example pipeline, in which a ci server listens for changes to a specific repo, then consumes anything it wants in that repo to run some tests and publish some stuff:


<http://concourse.ci/flight-school.html>

that example just listens to one repo and publishes one thing, but you can imagine consuming inputs from multiple repos, as per:

<http://concourse.ci/pipelines.html>

(the examples given are in concourse, but gocd or any other CI tool with pipeline semantics could accomplish the same thing)
